bmaptools/BmapHelpers.py

```diff
@@ -95,70 +93,61 @@ def program_is_available(name):
 
     return False
 
+def get_mount_point(path):
+    """Return the mount point for 'path'."""
 
-def get_mount_point(filepath):
-    """
-    Return the mount point the file object file resides on.
-    """
-    f = os.path.realpath(filepath)
-    while f != os.path.sep:
-        if os.path.ismount(f): return f
-        f = os.path.realpath(os.path.join(f, os.pardir))
-    return f
+    part = os.path.realpath(path)
+    while part != os.path.sep:
+        if os.path.ismount(part): 
+            return part
+        part = os.path.realpath(os.path.join(part, os.pardir))
+    return part
 
+def get_file_system_type(path):
+    """Return the file system type for 'path'."""
 
-def get_file_system_type(filepath):
-    """
-    Return the file system type the file object file resides on.
-    """
-
-    # normalize,resolve and get absolute path to the file object
-    f = os.path.realpath(filepath)
-    proc = subprocess.Popen(['df', '-T', '--', f], stdout=PIPE, stderr=PIPE)
+    # Normalize,resolve and get absolute path of the supplied path.
+    abspath = os.path.realpath(path)
+    proc = subprocess.Popen(['df', '-T', '--', abspath], stdout=PIPE, stderr=PIPE)
     stdout, stderr = proc.communicate()
+
+    # Parse the output of subprocess, for example:
+    # Filesystem                Type 1K-blocks     Used Available Use% Mounted on
+    # rpool/USERDATA/foo_5ucog2 zfs  456499712 86956288 369543424  20% /home/foo
+
     ftype = None
     if stdout:
-        t = stdout.splitlines()
-        if len(t) >= 2: 
-            t = t[1].split(None, 2)
-            if len(t) >= 2: 
-                ftype = t[1].lower()
+        lines = stdout.splitlines()
+        if len(lines) >= 2: 
+            fields = lines[1].split(None, 2)
+            if len(fields) >= 2: 
+                ftype = fields[1].lower()
 
     if not ftype:
-        raise Error("no file system type was found: %s" % stderr)
+        raise Error("no file system type was found for '%s': %s" % (path, stderr))
```

Comment:
I think it is better to include both stdout and stderr in the error message. Will be much more helpful in case this actually happen. How about something like this:

raise Error("failed to find file system type for path at '%s'.\nHere is the 'df -T' output.\nstdout:\n%s\nstderr:\n%s" % (path, stdout, stderr))